### ISSUE TITLE
DD_TRACE_ENABLED not being set for a specific function

### ISSUE THREAD

**Issue by sburnett-rbi** [DataDog/serverless-plugin-datadog#185]
## Expected Behavior

We have a repository with many lambda functions. Some of which are HTTP and others are event based. One particular function seems to have major problems when Datadog tracing is enabled. Previously, we were able to disable tracing for this specific function by setting an environment variable `DD_TRACE_ENABLED` to false in the `environment` section of that specific function. 
Expected behavior is that if a function has an environment variable set, that should be taken as precedent. It appears the datadog plugin is setting the env variable automatically and not considering that it is being explicitly set.

## Actual Behavior

Previously, this code would correctly set the environment variable to False.  What we are now seeing is that it is being set to 'true' for some reason.

## Steps to Reproduce the Problem

  1. Deploy the application to AWS
  2. View the environment variables

example serverless:
```
service: ${self:custom.stackName}

plugins:
  - serverless-pseudo-parameters
  - serverless-plugin-tracing
  - serverless-jetpack
  - serverless-plugin-disable-events
  - '@rbilabs/serverless-jetpack'
  - serverless-plugin-datadog

  datadog:
    addExtension: true
    apiKey: fakeApiKey
    site: fakeDDSite

functions:
    name: ${self:custom.stackName}-example-function
    timeout: 120
    handler: dist/functions/orders/index.exampleFunction
    events:
      - sqs:
          arn: exampleFunction
          batchSize: 5
    environment:
      DD_TRACE_ENABLED: 'false'
```

## Specifications

  - Serverless Framework version: `2.52.0`
  - Datadog Serverless Plugin version: `3.0.1`
  - Lambda function runtime (Python 3.7, Node 10, etc.):  `Node 14.17.3`
  
Really, I just want to know if there's a way to disable tracing for a single function. What I'm observing in the docs seems to indicate that you can only disable or enable tracing for the entire serverless stack, rather than an individual function.

**Comment by tianchu** [comment]
I see, that makes a lot of sense! It would require some changes and I would leave this issue open to assess the demand. 

You can probably workaround this issue by using the `exclude` option?

```yaml
custom:
  datadog:
    exclude:
      - functionToBeExcluded
```

This excludes this specific function from the plugin's magic. However, you can manually apply the same changes the plugin automates behind scene to this one particular function using serverless framework syntax. See the "custom" installation instructions https://docs.datadoghq.com/serverless/installation/nodejs?tab=custom for the exact changes that the plugin automates. Of course, you can leave `DD_TRACE_ENABLED` set to `false`.

**Comment by mrollinsiv** [comment]
I'm also running into this issue but cannot use the `exclude` workaround as I use the basic logging functionality on them. Additionally, doing the inverse creates a lot of unnecessary logic in the `serverless.yml` to simply enable tracing on all but a few.

**Comment by tianchu** [comment]
@astuyve should we reprioritize some improvements here? Also we might have similar overriding issue for other settings too? Perhaps we should tackle them all together and document how to override settings on function-level, which seems to be a popular use case?

**Comment by mrollinsiv** [comment]
That sounds great. I would think that any setting we define at the function level would override the same setting in a higher scope.

**Comment by astuyve** [comment]
Yes, that's the configuration paradigm set by the Serverless Framework, and therefore should be what we honor here. Extending that to all configuration options will take some time and require a breaking change, which is why this has been on hold for a bit. But we are going to prioritize this after wrapping up our current work. Thanks everyone!

**Comment by astuyve** [comment]
This should be fixed with #236. We'll do a new release with this fix shortly. Thanks!
